tests/samples.ts

```diff
@@ -215,3 +139,118 @@ export const albMultiValHeadersRequest = (): ApplicationLoadBalancerRequestEvent
       },
    });
 };
+
+export const makeAPIGatewayRequestContextIdentity = (): APIGatewayEventIdentity => {
+   return {
+      accessKey: null,
+      accountId: null,
+      apiKey: null,
+      apiKeyId: null,
+      caller: null,
+      clientCert: null,
+      cognitoAuthenticationProvider: null,
+      cognitoAuthenticationType: null,
+      cognitoIdentityId: null,
+      cognitoIdentityPoolId: null,
+      principalOrgId: null,
+      sourceIp: '12.12.12.12',
+      user: null,
+      userAgent: 'curl/7.54.0',
+      userArn: null,
+   };
+};
+
+export const makeAPIGatewayRequestContext = (params?: MakeAPIGatewayRequestEventContextParams): APIGatewayEventRequestContext => {
+   return {
+      accountId: '123456789012',
+      apiId: 'someapi',
+      authorizer: null,
+      httpMethod: params?.httpMethod ?? 'GET',
+      path: '/prd',
+      protocol: 'HTTP/1.1',
+      stage: 'prd',
+      requestId: 'a507736b-259e-11e9-8fcf-4f1f08c4591e',
+      requestTimeEpoch: 1548969891530,
+      resourceId: 'reas23acc',
+      identity: makeAPIGatewayRequestContextIdentity(),
+      resourcePath: '/',
+   };
+};
+
+export const makeAPIGatewayRequestEventHeaders =
+   (params?: MakeAPIGatewayRequestEventHeadersParams): APIGatewayRequestEvent['headers'] => {
+      return withDefault<APIGatewayRequestEvent['headers']>(
+         params,
+         {
+            Accept: '*/*',
+            'CloudFront-Forwarded-Proto': 'https',
+            'CloudFront-Is-Desktop-Viewer': 'true',
+            'CloudFront-Is-Mobile-Viewer': 'false',
+            'CloudFront-Is-SmartTV-Viewer': 'false',
+            'CloudFront-Is-Tablet-Viewer': 'false',
+            'CloudFront-Viewer-Country': 'US',
+            Host: 'b5gee6dacf.execute-api.us-east-1.amazonaws.com',
+            'User-Agent': 'curl/7.54.0',
+            Via: '2.0 4ee511e558a0400aa4b9c1d34d92af5a.cloudfront.net (CloudFront)',
+            'X-Amz-Cf-Id': 'xn-ohXlUAed-32bae2cfb7164fd690ffffb87d36b032==',
+            'X-Amzn-Trace-Id': 'Root=1-4b5398e2-a7fbe4f92f2e911013cba76b',
+            'X-Forwarded-For': '8.8.8.8, 2.3.4.5',
+            'X-Forwarded-Port': '443',
+            'X-Forwarded-Proto': 'https',
+            Referer: 'https://en.wikipedia.org/wiki/HTTP_referer',
+            Cookie: 'uid=abc; ga=1234; foo=bar; baz=foo%5Ba%5D; obj=j%3A%7B%22abc%22%3A123%7D; onechar=j; bad=j%3A%7Ba%7D',
+         }
+      );
```

Comment:
What if the developer only wants to edit a single header?
```suggestion
      return compact({
         Accept: '*/*',
         'CloudFront-Forwarded-Proto': 'https',
         'CloudFront-Is-Desktop-Viewer': 'true',
         'CloudFront-Is-Mobile-Viewer': 'false',
         'CloudFront-Is-SmartTV-Viewer': 'false',
         'CloudFront-Is-Tablet-Viewer': 'false',
         'CloudFront-Viewer-Country': 'US',
         Host: 'b5gee6dacf.execute-api.us-east-1.amazonaws.com',
         'User-Agent': 'curl/7.54.0',
         Via: '2.0 4ee511e558a0400aa4b9c1d34d92af5a.cloudfront.net (CloudFront)',
         'X-Amz-Cf-Id': 'xn-ohXlUAed-32bae2cfb7164fd690ffffb87d36b032==',
         'X-Amzn-Trace-Id': 'Root=1-4b5398e2-a7fbe4f92f2e911013cba76b',
         'X-Forwarded-For': '8.8.8.8, 2.3.4.5',
         'X-Forwarded-Port': '443',
         'X-Forwarded-Proto': 'https',
         Referer: 'https://en.wikipedia.org/wiki/HTTP_referer',
         Cookie: 'uid=abc; ga=1234; foo=bar; baz=foo%5Ba%5D; obj=j%3A%7B%22abc%22%3A123%7D; onechar=j; bad=j%3A%7Ba%7D',
         ...params,
      });
```

Reply:
@onebytegone I like this approach. I've also left a question about [applying it for the base object](https://github.com/silvermine/lambda-express/pull/91#discussion_r1893005957), what do you think?